mrbgems/mruby-string-ext/mrblib/string.rb

```diff
@@ -402,7 +402,7 @@ def upto(max, exclusive=false, &block)
       e = max.ord
       while c <= e
         break if exclusive and c == e
-        yield c.chr(__ENCODING__)
+        yield c.chr("☺".size == 1 ? "UTF-8" : "ASCII-8BIT")
```

Comment:
Not desirable, but changes will result in dynamic dispatch.
Perhaps it is preferable to change the meaning of the `__ENCODING__` pseudo-variables to get the run-time encoding, or add a constant to get the run-time encoding.